packages/@react-aria/selection/src/useTypeSelect.ts

```diff
@@ -37,14 +37,19 @@ export function useTypeSelect(options: TypeSelectOptions): TypeSelectAria {
       return;
     }
 
+    const isTimedOut = state.search === '';
+
     state.search += character;
 
     // Use the delegate to find a key to focus.
     // Prioritize items after the currently focused item, falling back to searching the whole list.
-    let key = keyboardDelegate.getKeyForSearch(state.search, selectionManager.focusedKey);
-    if (!key) {
-      key = keyboardDelegate.getKeyForSearch(state.search);
-    }
+    // If search times out, ignore the currently focused item (if any) and search from the top.
+    let key = isTimedOut 
+      ? keyboardDelegate.getKeyForSearch(state.search)
+      : keyboardDelegate.getKeyForSearch(state.search, selectionManager.focusedKey);
```

Comment:
I think we should always search starting from the focused key at the time the search started. I think the issue with my current implementation is that it searches starting from the focused key at the time each character is pressed, which changes as you type. If instead it only reset the location is searched from when the search timed out, then it would be more consistent and you'd still be able to find the next item after the current one easily. Let me know if that doesn't make sense.

Reply:
Yes, makes absolute sense. In fact, that's how I had it initially. But as I was comparing the behavior against `<select>` on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select), I realized that it wraps around after the timeout. Maybe we don't have to make it mimic `<select>`? I'm happy to make the change either way.

Reply:
I'd be happy with starting at the focused item.

Reply:
Working on it.